libs/ballot-interpreter-nh/src/image_utils.rs

```diff
@@ -227,6 +230,28 @@ pub fn find_scanned_document_inset(image: &GrayImage, threshold: u8) -> Option<I
     }
 }
 
+/// Calculates the degree to which the given image matches the given template,
+/// where 0.0 means the images are perfect inverses of each other and 1.0 means
+/// the images are identical. Panics if the image and template do not have the
+/// same dimensions.
+pub fn match_template<
+    I: GenericImageView<Pixel = Luma<u8>>,
+    T: GenericImageView<Pixel = Luma<u8>>,
+>(
+    image: &I,
+    template: &T,
+) -> UnitIntervalValue {
+    assert_eq!(image.dimensions(), template.dimensions());
+
+    let mut diff = 0.0;
+    for (image_pixel, template_pixel) in image.pixels().zip(template.pixels()) {
+        let image_luma = (image_pixel.2[0]) as f32;
+        let template_luma = (template_pixel.2[0]) as f32;
+        diff += (image_luma - template_luma).abs();
+    }
+    1.0 - (diff / (image.width() * image.height()) as f32 / f32::from(u8::MAX))
```

Comment:
What does `/ f32::from(u8::MAX))` do here?

Reply:
We're adding up the luminance value differences with a maximum per pixel of 255 aka `u8::MAX`, so to compute a score between `0.0` and `1.0` we divide by `u8::MAX`.